### PR TITLE
workspace: bump xla to enabled batching in gather

### PR DESCRIPTION
**Pull request by gwenzek**
Since we were using an "old" version of XLA we where missing the batching dims stablehlo attribute.
This was preventing the usage of batching in the gather operator.

Since I had already wrote the batching test before hand and was just ignoring the compilation error,
this PR just bump xla and remove the error suppression code in tests.

### CODE REVIEW

**Comment by gwenzek** [comment]
it could also help #19 which was using a Cuda 560.

File: examples/MODULE.bazel.lock
Comment (gwenzek) [inline examples/MODULE.bazel.lock:1]: feels weird, should I keep that ?

Comment (steeve) [reply examples/MODULE.bazel.lock:1]: yes, some extensions might need to be marked as reproducible, but are not yet :(

File: third_party/modules/stablehlo/metadata.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
         "github:openxla/stablehlo"
     ],
     "versions": [
-        "20240829.0-54aa1a5"
+        "20240829.0-54aa1a5",
+        "20240917.0-78c753a",
```
Comment (steeve) [inline third_party/modules/stablehlo/metadata.json:15]: glad to see trailing commas are working !

**Review by steeve — CHANGES_REQUESTED** [review]
quick question on tsl.bzl and xla.bzl but else looks good!

File: third_party/modules/llvm-raw/20240919.0-94c024a/MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module(
+    name = "llvm-raw",
+    version = "20240919.0-94c024a",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "platforms", version = "0.0.10")
+bazel_dep(name = "zstd", version = "1.5.6", repo_name = "llvm_zstd")
```
Comment (steeve) [inline third_party/modules/llvm-raw/20240919.0-94c024a/MODULE.bazel:9]: same for those versions

File: third_party/modules/xla/20240919.0-1b18dd6/overlay/workspace.bzl [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+load("@tsl//tools/toolchains/remote:configure.bzl", "remote_execution_configure")
+
+def _xla_workspace_impl(mctx):
+    cuda_configure(name = "local_config_cuda")
+    remote_execution_configure(name = "local_config_remote_execution")
+    rocm_configure(name = "local_config_rocm")
+    tensorrt_configure(name = "local_config_tensorrt")
+    tf_http_archive(
+        name = "com_github_grpc_grpc",
+        sha256 = "b956598d8cbe168b5ee717b5dafa56563eb5201a947856a6688bbeac9cac4e1f",
```
[the remark this answers is not in the document]
Comment (gwenzek) [reply third_party/modules/xla/20240919.0-1b18dd6/overlay/workspace.bzl:14]: yes, protobuf and grpc didn't change version.

File: third_party/modules/llvm-raw/20240919.0-94c024a/MODULE.bazel (hunk 2 above)
Comment (gwenzek) [reply third_party/modules/llvm-raw/20240919.0-94c024a/MODULE.bazel:9]: and llvm seems to be using zstd 1.3.0 https://github.com/llvm/llvm-project/blob/6e6d5eae765939cc4074bdd606658e78c4a2a559/utils/bazel/WORKSPACE#L131

**Review by steeve — APPROVED** [review]

**Pull request merged**